beater/handlers.go

```diff
@@ -96,8 +96,9 @@ func frontendHandler(pf ProcessorFactory, config *Config, report reporter) http.
 		logp.Err(err.Error())
 	}
 	prConfig := processor.Config{
-		SmapMapper:     smapper,
-		LibraryPattern: regexp.MustCompile(config.Frontend.LibraryPattern),
+		SmapMapper:          smapper,
+		LibraryPattern:      regexp.MustCompile(config.Frontend.LibraryPattern),
+		ExcludeFromGrouping: regexp.MustCompile(config.Frontend.ExcludeFromGrouping),
```

Comment:
I didn't notice that we were doing this already with `LibraryPattern` but are we ok with a panic if a bad regex is used?  That seems too unfriendly for GA but maybe ok for alpha frontend support?

Reply:
Maybe something like:
```
diff --git a/beater/beater.go b/beater/beater.go
index bce4eac..feb0124 100644
--- a/beater/beater.go
+++ b/beater/beater.go
@@ -4,6 +4,7 @@ import (
 	"fmt"
 	"net"
 	"net/http"
+	"regexp"
 
 	"github.com/elastic/beats/libbeat/beat"
 	"github.com/elastic/beats/libbeat/common"
@@ -26,6 +27,16 @@ func New(b *beat.Beat, ucfg *common.Config) (beat.Beater, error) {
 		beaterConfig.setElasticsearch(b.Config.Output.Config())
 	}
 
+	if beaterConfig.Frontend.isEnabled() {
+		if _, err := regexp.Compile(beaterConfig.Frontend.LibraryPattern); err != nil {
+			return nil, err
+		}
+
+		if _, err := regexp.Compile(beaterConfig.Frontend.ExcludeFromGrouping); err != nil {
+			return nil, err
+		}
+	}
+
 	bt := &beater{
 		config: beaterConfig,
 	}
```

But with errors wrapped with helpful messaging.